sentry-core/src/main/java/io/sentry/core/transport/HttpTransport.java

```diff
@@ -161,16 +161,49 @@ public HttpTransport(
    */
   @Override
   public boolean isRetryAfter(final @NotNull String type) {
-    if (sentryRetryAfterLimit.containsKey(type)) {
-      final Date date = sentryRetryAfterLimit.get(type);
+    final String category = getCategoryFromType(type);
+    final Date currentDate = new Date();
+
+    // check all categories
+    if (sentryRetryAfterLimit.containsKey(HTTP_RETRY_ALL_CATEGORIES)) {
+      final Date date = sentryRetryAfterLimit.get(HTTP_RETRY_ALL_CATEGORIES);
+      if (!currentDate.after(date)) {
+        return true;
+      }
+    }
+    // check for specific category
+    if (sentryRetryAfterLimit.containsKey(category)) {
+      final Date date = sentryRetryAfterLimit.get(category);
+      return !currentDate.after(date);
+    }
 
-      return !new Date().after(date);
-    } else if (sentryRetryAfterLimit.containsKey(HTTP_RETRY_DEFAULT_CATEGORY)) {
-      final Date date = sentryRetryAfterLimit.get(HTTP_RETRY_DEFAULT_CATEGORY);
+    return false;
+  }
+
+  /**
+   * Returns a rate limiting category from type
+   *
+   * @param type the type (eg event, session, attachment, ...)
+   * @return the category eg (error, session, attachment)
+   */
+  private @NotNull String getCategoryFromType(final @NotNull String type) {
+    // fallback everything else to 'default' category
+    String category = "default";
```

Comment:
this can be improved, yes.

but a downside of an enum in java is, it cant have a public ctor so it cant be used outside of the package and we do use packages for the organization.
We have `SentryEnvelopeItemType` already.

I'd improve it in another PR if thats ok